### PR TITLE
Freq problem v2

### PR DESCRIPTION
**Pull request by MateusGheorghe**

### CODE REVIEW

**Comment by MateusGheorghe** [comment]
Still needs to solve business day problem

File: neuralprophet/forecaster.py [hunk 1]
```diff
@@ -805,16 +805,17 @@ def split_df(self, df, freq, valid_p=0.2, local_modeling=False):
...
+                Any valid frequency for pd.date_range, such as '5min', 'D', 'MS' or 'auto' (default) to automatically set frequency.
             valid_p (float): fraction of data to use for holdout validation set
                 Targets will still never be shared.
 
         Returns:
             df_train (pd.DataFrame):  training data
             df_val (pd.DataFrame): validation data
         """
-        df = df.copy(deep=True)
+        df = df.copy()
```
Comment (MateusGheorghe) [inline neuralprophet/forecaster.py:816]: @ourownstory This should be solved by the df copy PR. I had to change it here too so the Split df feature could work.

**Comment by MateusGheorghe** [comment]
@ourownstory guess this is ready to be merged

**Comment by ourownstory** [comment]
final todos: 
- docstrings
- 98% test coverage of _infer_freq

**Pull request merged**